gcloud/datastore/helpers.py

```diff
@@ -45,12 +46,17 @@ def entity_from_protobuf(pb):
     :returns: The entity derived from the protobuf.
     """
     key = key_from_protobuf(pb.key)
-    entity = Entity(key=key)
+    entity_props = {}
+    exclude_from_indexes = []
 
     for property_pb in pb.property:
         value = _get_value_from_property_pb(property_pb)
-        entity[property_pb.name] = value
+        entity_props[property_pb.name] = value
+        if not property_pb.value.indexed:
```

Comment:
I agree. Proposed solution:

I check to make sure that all indexed values agree (all `False` or all `True`) and throw an exception if they don't.

Potential sharp corners:

Certain things like an `Entity` pb, `bytes` over 500 chars in length, etc. can't be indexed, so if we introduced behavior to detect and turn this off, we'd cause our own "disagreements".

----

No matter how we slice it, mapping to the [`Value`][1] message class without actually using it is hard.

[1]: https://github.com/GoogleCloudPlatform/google-cloud-datastore/blob/21e029b2657ebb15fef5429c90c4315326855fd1/proto/datastore_v1.proto#L99